ReactAndroid/src/main/java/com/facebook/react/devsupport/ReactInstanceDevCommandsHandler.java

```diff
@@ -31,4 +35,10 @@
    * Request to toggle the react element inspector.
    */
   void toggleElementInspector();
+
+  /**
+   * Return activity react views are mounted in. This will be used by DevSupportManager to launch
+   * dialogs (such as dev settings dialog or a red box)
+   */
+  @Nullable Activity getCurrentActivity();
```

Comment:
Would there be another way to pass around the activity? It feels kind of out of place on this interface.

Reply:
I agree, on the other hand new interface would complicate this part of the codebase even more which I don't think is necessary considering it doesn't do much.

Which of these two options would you prefer:
1) Rename this interface so that this method fits better
2) This interface implementation just passes all the calls to private methods of ReactInstanceManager. I could make those methods public and instead of passing `ReactInstanceDevCommandsHandler` ref to `DevSupportManagerFactory` I could pass ref to `ReactInstanceManager`

PS: I prefer the latter

Reply:
Yea if it doesn't require too many changes I think number 2 would be nice

Reply:
@janicduplessis updated this PR with this change. Note that now it is more likely that this code will break some internal fb stuff where they perhaps use their own implementation of devsupportmanager